core/src/Cuda/Kokkos_Cuda_GraphNodeKernel.hpp

```diff
@@ -90,13 +84,23 @@ class GraphNodeKernelImpl<Kokkos::Cuda, PolicyType, Functor, PatternTag,
   cudaGraphNode_t* get_cuda_graph_node_ptr() const { return m_graph_node_ptr; }
   cudaGraph_t const* get_cuda_graph_ptr() const { return m_graph_ptr; }
 
-  Kokkos::ObservingRawPtr<base_t> allocate_driver_memory_buffer() const {
+  Kokkos::ObservingRawPtr<base_t> allocate_driver_memory_buffer(
+      const CudaSpace& mem) const {
     KOKKOS_EXPECTS(m_driver_storage == nullptr)
-    m_driver_storage = static_cast<base_t*>(Kokkos::CudaSpace().allocate(
-        "GraphNodeKernel global memory functor storage", sizeof(base_t)));
+    m_driver_storage = std::shared_ptr<base_t>(
```

Comment:
Do we need the template parameter here?

Reply:
Probably not.

Reply:
In fact, it is needed (at least for `hipcc`):
```bash
/workspaces/kokkos/core/src/HIP/Kokkos_HIP_GraphNodeKernel.hpp:73:24: error: no viable constructor or deduction guide for deduction of template arguments of 'shared_ptr'
   73 |     m_driver_storage = std::shared_ptr(static_cast<base_t*>(Kokkos::HIPSpace().allocate(exec,
```